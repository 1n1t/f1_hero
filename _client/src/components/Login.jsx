import React from 'react';

const Login = () => (
  <div className="login">
    <h1>F1 Hero</h1>
    <p>be number 1 in formula 1 world</p>
    <a href="/auth/twitter" className="btn">
      Login <i className="fa fa-twitter" aria-hidden="true" />
    </a>
  </div>
);

export default Login;
