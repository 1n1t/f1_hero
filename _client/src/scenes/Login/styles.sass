.Login
  background-color: $color-black
  color: white
  text-align: center
  padding: 40px 50px
  display: flex
  flex-direction: column
  justify-content: space-between
  align-items: center
  width: 300px
  height: 400px
  position: absolute
  left: calc(50% - 200px)
  top: calc(50% - 240px)

  h1
    font-size: 3rem
    text-transform: uppercase

  p
    padding: 0 100px
    line-height: 1.7rem
    margin-bottom: 50px

  .btn
    border: 3px solid white
    .fa
      margin-left: 20px
