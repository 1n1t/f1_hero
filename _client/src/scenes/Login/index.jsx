import React from 'react';

import './styles.scss';

const Login = () => (
	<div className="Login">
		<h1>F1 Hero</h1>
		<p>be number 1 in formula 1 world</p>
		<a href="/auth/twitter" className="btn">
			Login <i className="fa fa-twitter" aria-hidden="true" />
		</a>
	</div>
);

export default Login;
