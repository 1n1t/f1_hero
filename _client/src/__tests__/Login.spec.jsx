import React from 'react';
import Login from '../components/Login';
import {shallow} from 'enzyme';

test('Login renders correctly', () => {
  const component = shallow(<Login/>);
  expect(component).toMatchSnapshot();
});