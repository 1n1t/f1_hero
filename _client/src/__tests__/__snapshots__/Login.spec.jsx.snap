// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="login"
  >
    <h1>
      F1 Hero
    </h1>
    <p>
      be number 1 in formula 1 world
    </p>
    <a
      className="btn"
      href="/auth/twitter"
    >
      Login 
      <i
        aria-hidden="true"
        className="fa fa-twitter"
      />
    </a>
  </div>,
  "nodes": Array [
    <div
      className="login"
    >
      <h1>
        F1 Hero
      </h1>
      <p>
        be number 1 in formula 1 world
      </p>
      <a
        className="btn"
        href="/auth/twitter"
      >
        Login 
        <i
          aria-hidden="true"
          className="fa fa-twitter"
        />
      </a>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Login />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="login"
        >
          <h1>
            F1 Hero
          </h1>
          <p>
            be number 1 in formula 1 world
          </p>
          <a
            className="btn"
            href="/auth/twitter"
          >
            Login 
            <i
              aria-hidden="true"
              className="fa fa-twitter"
            />
          </a>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="login"
        >
          <h1>
            F1 Hero
          </h1>
          <p>
            be number 1 in formula 1 world
          </p>
          <a
            className="btn"
            href="/auth/twitter"
          >
            Login 
            <i
              aria-hidden="true"
              className="fa fa-twitter"
            />
          </a>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Login />,
}
`;
